tests/runtime/servicers/test_model_train_servicer_impl.py

```diff
@@ -185,38 +185,66 @@ def test_model_train_validation_error_raises(sample_model_train_servicer, output
 
 #####################################################################
 # Normal tests
-def test_model_train_sample_widget(sample_model_train_servicer, output_dir):
+@pytest.mark.parametrize("output_in_config", (True, False))
+def test_model_train_sample_widget(
+    sample_model_train_servicer,
+    output_dir,
+    output_in_config,
+):
+    """This test tests end-to-end training. It includes verifying that the model
+    is saved in the right place. The place where the model is saved comes from:
+
+    1. get_config().runtime.training.output_dir
+    2. request.training_output_dir
+
+    IN THAT ORDER!
```

Comment:
all the more reason that the `process_pb2` handler should move out of caikit/caikit!

Reply:
💯